src/Npgsql/TypeHandlers/ArrayHandler.cs

```diff
@@ -326,26 +344,36 @@ protected internal override Task WriteObjectWithLength(object value, NpgsqlWrite
 
             buf.WriteInt32(dimensions);
             buf.WriteInt32(1);  // HasNulls=1. Not actually used by the backend.
-            buf.WriteUInt32(ElementHandler.PostgresType.OID);
+            buf.WriteUInt32(_elementHandler.PostgresType.OID);
             if (asArray != null)
             {
                 for (var i = 0; i < dimensions; i++)
                 {
                     buf.WriteInt32(asArray.GetLength(i));
-                    buf.WriteInt32(LowerBound);  // We don't map .NET lower bounds to PG
+                    buf.WriteInt32(_lowerBound);  // We don't map .NET lower bounds to PG
                 }
             }
             else
             {
                 buf.WriteInt32(value.Count);
-                buf.WriteInt32(LowerBound);  // We don't map .NET lower bounds to PG
+                buf.WriteInt32(_lowerBound);  // We don't map .NET lower bounds to PG
             }
 
             foreach (var element in value)
-                await ElementHandler.WriteObjectWithLength(element, buf, lengthCache, null, async);
+                await _elementHandler.WriteObjectWithLength(element, buf, lengthCache, null, async);
         }
 
         #endregion
+
+        #region Misc
+        
+        protected internal override ArrayHandler CreateArrayHandler(PostgresType arrayBackendType)
+            => new ArrayHandler<short>(this) { PostgresType = arrayBackendType };
```

Comment:
@YohDeadfall @roji 
I just stumbled upon this while manually merging those changes for #443 and I just can't get my head around the fact that it's `ArrayHandler<short>(this)` .
Do we even support creating ArrayHandlers from ArrayHandler<TElement>?
Is there any use in returning anything else but `this` or possibly `ArrayHandler<TElement>(this)` ?
Do you have any pointers for me?

Reply:
Thanks for catching that! It's a bug, will fix.